### ISSUE TITLE
Outdated SESSION_MANAGER after relogin&attach

### ISSUE THREAD

**Issue by reverendhomer** [tmux/tmux#1494]
Hi,

I'm working with tmux 2.6 on Ubuntu 18.04. And I've caught an issue that can be reproduced by:
* Create new tmux session
* Run `vim` in it (it's important to have vim with X11 support, i.e. `vim-gtk3` package in ubuntu) — it will start immediately, without any delay
* Close vim, logout from the session *without closing terminal* and login back
* Run terminal, attach to the tmux session
* Try to run vim again

There will be a one-second-delay before vim started. Logs say vim has one-second long `xsmp init` (that's connection to the X server to enable X clipboard).

However, if you do `echo $SESSION_MANAGER` from the tmux session and from the terminal without tmux session after relogin, you will see they're different. If you update contents of the `$SESSION_MANAGER` with the new value from the terminal without tmux, vim will be starting immediately again.

I can reproduce that on Fedora 28 with tmux 2.7 as well. Also, this issue is reproducible with empty config.

My config:
```
λ uname -sp && tmux -V && echo $TERM
Linux x86_64
tmux 2.6
screen-256color
```

Log file (I reproduced all my steps): [tmux-server-13704.log](https://gist.githubusercontent.com/reverendhomer/c62658dafc14d44b7a2dedc532187885/raw/51499e361c242a9ec8c79ccc75ff8a5addbac63b/tmux-server-13704.log)

**Comment by nicm** [comment]
Did you read the section of the

**Comment by reverendhomer** [comment]
> Did you read the section of the manual about the environment?

Yes, I added 
```
set-option -ga update-environment ' SESSION_MANAGER'
```
to my `.tmux.conf`, but that didn't work.

**Comment by nicm** [comment]
update-environment can only work for new panes in existing tmux sessions, are you creating a new pane?

**Comment by reverendhomer** [comment]
> update-environment can only work for new panes in existing tmux sessions, are you creating a new pane?

Nope. Can I update `SESSION_MANAGER` in existing pane other way than `export SESSION_MANAGER=<...>`?

**Comment by nicm** [comment]
No, there is no sensible way for tmux to modify the environment of other processes.  You will need to do it from inside the pane, you can use tmux showenv to get the updated value if any.
